Spigot-Server-Patches/0425-Use-result-of-setItemStack-in-PlayerBucketEmptyEvent.patch

```diff
@@ -0,0 +1,53 @@
+From 210cf7b13fab9c83e8f3b10b80a8aea5a4e0affd Mon Sep 17 00:00:00 2001
+From: Draycia <lonelyyordle@gmail.com>
+Date: Tue, 14 Jan 2020 05:37:05 -0800
+Subject: [PATCH] Use result of setItemStack in PlayerBucketEmptyEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/EntityLiving.java b/src/main/java/net/minecraft/server/EntityLiving.java
+index 80219f2df..c8184f94e 100644
+--- a/src/main/java/net/minecraft/server/EntityLiving.java
++++ b/src/main/java/net/minecraft/server/EntityLiving.java
+@@ -1897,6 +1897,7 @@ public abstract class EntityLiving extends Entity {
+         }
+     }
+ 
++    public void setItemInHand(EnumHand enumhand, ItemStack itemstack) { this.a(enumhand, itemstack); } // Paper - OBFHELPER
+     public void a(EnumHand enumhand, ItemStack itemstack) {
+         if (enumhand == EnumHand.MAIN_HAND) {
+             this.setSlot(EnumItemSlot.MAINHAND, itemstack);
+diff --git a/src/main/java/net/minecraft/server/ItemBucket.java b/src/main/java/net/minecraft/server/ItemBucket.java
+index 0ff92aea5..498cd4d24 100644
+--- a/src/main/java/net/minecraft/server/ItemBucket.java
++++ b/src/main/java/net/minecraft/server/ItemBucket.java
+@@ -134,12 +134,13 @@ public class ItemBucket extends Item {
+             } else {
+                 // CraftBukkit start
+                 if (entityhuman != null) {
+-                    PlayerBucketEmptyEvent event = CraftEventFactory.callPlayerBucketEmptyEvent(world, entityhuman, blockposition, clicked, enumdirection, itemstack, enumhand);
++                    PlayerBucketEmptyEvent event = CraftEventFactory.callPlayerBucketEmptyEvent(world, entityhuman, blockposition, clicked, enumdirection, entityhuman.isCreative() ? entityhuman.getItemInHand(enumhand) : itemstack, enumhand); // Paper - 2651: Prevents creative players from having their buckets modified
+                     if (event.isCancelled()) {
+                         ((EntityPlayer) entityhuman).playerConnection.sendPacket(new PacketPlayOutBlockChange(world, blockposition)); // SPIGOT-4238: needed when looking through entity
+                         ((EntityPlayer) entityhuman).getBukkitEntity().updateInventory(); // SPIGOT-4541
+                         return false;
+                     }
++                    entityhuman.setItemInHand(enumhand, org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(event.getItemStack())); // Paper - 2651: set item in hand if event isn't cancelled
```

Comment:
this needs to be passed to the caller so it can set the item result in the interact wrapper - the wrapper is the proper way to handle changes